### PR TITLE
Backends tidyup

### PR DESCRIPTION
**Pull request by jcarrano**
This is a refactoring only PR to prepare for the multiple-backend support. The idea is to have file deletions and renames here so that the other diff, which will already have many changes, is not so huge.

I was able to test the Linux, Plain and Android versions. ~I had some trouble getting Windows to cooperate~ I got windows working, and I don't own a Mac to test the MacOS code.

Some of the code that is added here will later be removed with the backend changes.

### CODE REVIEW

**Review by kdewald — CHANGES_REQUESTED** [review]
FYI, the MacOS build is broken, see CI run for details.

Loved the idea of the Builder template class! I do however have a small request: Can we use a more explicit naming or namespacing for the `build`, `build_vec` and `collect_values` functions to prevent them from being this generic?

Something similar to the `CollectValues` class, could we name it `ValueCollector` instead?

**Comment by kdewald** [comment]
The `clang-format` and `cppcheck` tests are currently failing (mostly from ongoing Android work), but please have a quick look to validate that nothing new is being added.

File: simpleble/src/builders/BuildVec.h [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 #include "BuilderBase.h"
 
-namespace SimpleBLE {
+namespace SimpleBLE::builder {
```
Comment (kdewald) [inline simpleble/src/builders/BuildVec.h:5]: I'll be slightly pedantic here, can we use `Factory` as the namespace?

File: simpleble/src/builders/BuilderBase.h [hunk 2]
```diff
@@ -5,7 +5,7 @@
 #include <type_traits>
 #include <utility>
 
-namespace SimpleBLE {
+namespace SimpleBLE::builder {
```
Comment (kdewald) [inline simpleble/src/builders/BuilderBase.h:8]: I'll be slightly pedantic here, can we use `Factory` as the namespace?

File: simpleble/src/backends/macos/PeripheralMac.mm [hunk 3]
```diff
@@ -1,6 +1,7 @@
 #import "PeripheralMac.h"
-#import "PeripheralBaseMacOS.h"
 #import "BuilderBase.h"
+#import "PeripheralBaseMacOS.h"
+#include "ServiceBase.h"
```
Comment (kdewald) [inline simpleble/src/backends/macos/PeripheralMac.mm:4]: Objective-C files should use `import`

File: simpleble/src/CommonUtils.h [hunk 4]
```diff
@@ -26,10 +26,10 @@
         }                                                                                   \
     } while (0)
 
-namespace SimpleBLE {
+namespace SimpleBLE::util {
```
Comment (kdewald) [inline simpleble/src/CommonUtils.h:29]: I'll be slightly pedantic here, can we use `Util` as the namespace?

File: simpleble/src/CommonUtils.h [hunk 5]
```diff
@@ -46,8 +46,8 @@ struct CollectValues {
  * Collect the values of a mapping like object into a vector-like object.
  */
 template <typename MAP>
-auto collect_values(const MAP& map) {
-    return CollectValues<MAP>{map};
+auto values(const MAP& map) {
```
Comment (kdewald) [inline simpleble/src/CommonUtils.h:49]: I think `collect_values` is correct if within the `Util` namespace

File: simpleble/src/backends/macos/AdapterMac.mm [hunk 6]
```diff
@@ -1,14 +1,15 @@
 #import "AdapterMac.h"
 #import "AdapterBaseMacOS.h"
+#import "BuildVec.h"
+#import "BuilderBase.h"
 #import "CommonUtils.h"
+#include "CommonUtils.h"
```
Comment (kdewald) [inline simpleble/src/backends/macos/AdapterMac.mm:6]: Duplicate import

File: simpleble/src/builders/BuildVec.h [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 #include "BuilderBase.h"
```
Comment (kdewald) [inline simpleble/src/builders/BuildVec.h:3]: I think it makes sense to roll this file into `BuilderBase.h`

File: simpleble/src/backends/macos/PeripheralBaseMacOS.mm [hunk 8]
```diff
@@ -3,7 +3,10 @@
 #import "LoggingInternal.h"
 #import "Utils.h"
 
+#import <simpleble/Descriptor.h>
```
Comment (kdewald) [inline simpleble/src/backends/macos/PeripheralBaseMacOS.mm:6]: One of the imports is incorrect.

**Comment by jcarrano** [comment]
The cppcheck is failing because it crashes, but when I run it locally it succeeds. Clang-format is complaining about files I did not touch.

**Comment by kdewald** [comment]
> The cppcheck is failing because it crashes, but when I run it locally it succeeds. Clang-format is complaining about files I did not touch.

Yeah, a lot of that comes from the changes we're doing in the Android side, so no need to worry about that for now.

**Review by kdewald — APPROVED** [review]

**Pull request merged**